pysph/tools/ipython.py

```diff
@@ -0,0 +1,641 @@
+        cache             Cached data
+
+    Methods
+    -------
+
+        get_buffer
+        show_log
+        show_info
+        show_results
+        show_all
+
+    '''
+
+    def __init__(self, path, cache=True):
+
+        # Imports #
+
+        from pysph.solver.utils import get_files, load
+
+        # Configuring the path #
+
+        self.path = path
+        self.paths_list = get_files(path)
+
+        # Caching #
+        # Note : Caching is only used by get_buffer and widget handlers.
+        if cache:
+            self.cache = {}
+        else:
+            pass
+
+    def get_buffer(self, frame):
+        '''Return particle arrays for a given frame number with caching.
+
+
+        Parameters
+        ----------
+
+        frame : int
+
+        Returns
+        -------
+
+        A dictionary.
+
+        Examples
+        --------
+
+        >>> sample = Viewer2D('/home/deep/pysph/trivial_inlet_outlet_output/')
+        >>> sample.get_buffer(12)
+        {'arrays': {'fluid': <pysph.base.particle_array.ParticleArray at 0x7f3f7d144d60>,
+        'inlet': <pysph.base.particle_array.ParticleArray at 0x7f3f7d144b98>,
+        'outlet': <pysph.base.particle_array.ParticleArray at 0x7f3f7d144c30>},
+        'solver_data': {'count': 240, 'dt': 0.01, 't': 2.399999999999993}}
+
+
+        '''
+        if hasattr(self, "_cbars"):
+            if frame in self.cache.keys():
+                temp_data = self.cache[frame]
+            else:
+                from pysph.solver.utils import load
+                self.cache[frame] = temp_data = load(self.paths_list[frame])
+        else:
+            from pysph.solver.utils import load
+            temp_data = load(self.paths_list[frame])
+
+        return temp_data
+
+    def show_log(self):
+        '''
+        Prints the content of log file.
+        '''
+        import glob
+        print("Printing log : \n\n")
+        path = self.path + "*.log"
+        logfile = open(glob.glob(path)[0], 'r')
+        for lines in logfile:
+            print(lines)
+        logfile.close()
+
+    def show_results(self):
+        '''
+        Show if there are any png, jpeg, jpg, or bmp images.
+        '''
+
+        import glob
+        from IPython.display import Image, display
+
+        imgs = tuple()
+        for extension in ['png', 'jpg', 'jpeg', 'bmp']:
+            temppath = self.path + "*." + extension
+            for paths in glob.glob(temppath):
+                imgs += (Image(paths),)
+        if len(imgs) != 0:
+            display(*imgs)
+        else:
+            print("No results to show.")
+
+    def show_info(self):
+        '''
+        Print contents of the .info file present in the output directory,
+        keys present in results.npz,  number of files and
+        information about paricle arrays.
+
+        Examples
+        --------
+
+        >>> sample = Viewer2D('/home/deep/pysph/trivial_inlet_outlet_output/')
+        >>> sample.show_info()
+        Printing info :
```

Comment:
I think this example is not really needed here, it just makes for a long docstring without really adding too much value.